src/native/fs/fs_napi.cpp

```diff
@@ -1,46 +1,124 @@
+        if (fd < 0) {                                                           \
+            SetError(XSTR() << _desc << ": ERROR not opened " << _wrap->_path); \
+            return;                                                             \
+        }                                                                       \
+    } while (0)
+
+// This macro should be used after calling any syscall/gpfs-syscall that uses errno
+// in case of failures - it will throw the errno using SetSyscallError
+#define SYSCALL_OR_RETURN(x)   \
+    do {                       \
+        int r = (x);           \
+        if (r) {               \
+            SetSyscallError(); \
+            return;            \
+        }                      \
+    } while (0)
+
+// This macro should be used after calling any gpfs-fcntl which doesn't use errno
+// It will handle error which number will be saved under gpfs_error
+#define GPFS_FCNTL_OR_RETURN(x)                                              \
+    do {                                                                     \
+        int gpfs_error = 0;                                                  \
+        int r = (x);                                                         \
+        if (r) {                                                             \
+            if (gpfs_error) {                                                \
+                SetError(XSTR() << "GPFS FCNTL error:" << DVAL(gpfs_error)); \
+            } else {                                                         \
+                SetSyscallError();                                           \
+            }                                                                \
+            return;                                                          \
+        }                                                                    \
+    } while (0)
+
+// This macro should be used when calling fstat in case we want to validate
+// no race condition happened changing the fd during the function run
+#define CHECK_CTIME_CHANGE(fd, stat_res, path)                    \
+    do {                                                          \
+        auto start_ctime = stat_res.st_ctime;                     \
+        SYSCALL_OR_RETURN(fstat(fd, &stat_res));                  \
+        auto end_ctime = stat_res.st_ctime;                       \
+        if (start_ctime != end_ctime) {                           \
+            SetError(XSTR() << "FileStat: " << DVAL(path)         \
+                            << " cancelled due to ctime change"); \
+        }                                                         \
+    } while (0)
+
+#ifdef __APPLE__
+#define flistxattr(a, b, c) ::flistxattr(a, b, c, 0)
+#define getxattr(a, b, c, d) ::getxattr(a, b, c, d, 0, 0)
+#define fgetxattr(a, b, c, d) ::fgetxattr(a, b, c, d, 0, 0)
+#define fsetxattr(a, b, c, d, e) ::fsetxattr(a, b, c, d, e, 0)
+#define fremovexattr(a, b) ::fremovexattr(a, b, 0)
```

Comment:
Did the indentation slip? 

True in all cases if `#ifdef __APPLE__` in this file

Reply:
no it didn't - by default preprocessor directives are not indented. We just formatted the code with code formatter and it fixed the indent to be on first column.